File: src/excise01/Ex01.java
package excise01;

public class Ex01 {
    public static void main(String[] args) {
        int[] array = parceToOctal(15);
        System.out.print("Octal: ");
        for (int e : array) {
            System.out.print(e);
        }
    }

    public static int[] parceToOctal(int decimal) {
        int[] octal = {0, 0, 0, 0};
        int index = 0;
        while (decimal > 0) {
            octal[octal.length - 1 - index] = decimal % 8;
            decimal /= 8;
            index++;
        }
        return octal;
    }
}

File: src/excise01/Ex01.kt
package excise01

fun main(args: Array<String>) {
    val array : Array<Int> = parceToOctal(15)
    print("Octal: ")
    for (elem in array) {
        print(elem)
    }
    println()
}

fun parceToOctal(d : Int): Array<Int> {
    var decimal : Int = d
    val octal: Array<Int> = arrayOf(0, 0, 0, 0)
    var index: Int = 0
    while (decimal > 0) {
        octal[octal.size - 1 - index] = decimal % 8
        decimal /= 8
        index++
    }
    return octal
}